templates/pkg/resource/manager.go.tpl

```diff
@@ -156,6 +162,69 @@ func (rm *resourceManager) ARNFromName(name string) string {
 	)
 }
 
+// LateInitialize returns an acktypes.AWSResource after setting the late initialized
+// fields from the readOne call. This method will initialize the optional fields
+// which were not provided by the k8s user but were defaulted by the AWS service.
+// If there are no such fields to be initialized, the returned object is similar to
+// object passed in the parameter.
+func (rm *resourceManager) LateInitialize(
+	ctx context.Context,
+	latest acktypes.AWSResource,
+) (acktypes.AWSResource, error) {
+	rlog := ackrtlog.FromContext(ctx)
+	// If there are no fields to late initialize, do nothing
+	if len(lateInitializeFieldNames) == 0 {
+		rlog.Debug("no late initialization required.")
+		return latest, nil
+	}
+	lateInitConditionReason := ""
+	lateInitConditionMessage := ""
+{{- if $hookCode := Hook .CRD "late_initialize_pre_read_one" }}
+{{ $hookCode }}
+{{- end }}
+	observed, err := rm.ReadOne(ctx, latest)
+	if err != nil {
+		lateInitConditionMessage = "Unable to complete Read operation required for late initialization"
+		lateInitConditionReason = "Late Initialization Failure"
+		ackcondition.SetLateInitialized(latest, corev1.ConditionFalse, &lateInitConditionMessage, &lateInitConditionReason)
+		return latest, ackrequeue.NeededAfter(err, time.Duration(0)*time.Second)
```

Comment:
This should probably be `return latest, err`. The `handleReconcileError` in the common runtime will handle requeues if necessary and by triggering an immediate requeue here if the ReadOne call errored is just going to cause an immediate call to ReadOne again (in the requeue'd reconcile event) which would likely just return the exact same error.